UX: Improve "Cannot read property 'forEach' of undefined" error msg

Currently the output of `Cannot read property 'forEach' of undefined` is really hard for beginners to diagnose. This can be caused by a number of things and it would be nice to either have the docs cover it or to improve the error message.

Causes
- When passing in invalid components to PolymerElement like `PolymerElement('thanks-vaadin')`
- Not importing the component via something like `<link rel="import" href="bower_components/iron-icons/av-icons.html">`
- Probably some others...

Full stack trace:

``` javascript
zone.js:461 Unhandled Promise rejection: TypeError: Cannot read property 'forEach' of undefined
        at Object.PolymerElement (http://localhost:4200/vendor/@vaadin/angular2-polymer/src/polymer-element.js:71:20)
        at eval (http://localhost:4200/app/+landing/landing.component.js:41:36)
        at Object.eval (http://localhost:4200/app/+landing/landing.component.js:48:2)
        at eval (http://localhost:4200/app/+landing/landing.component.js:51:4)
        at eval (http://localhost:4200/app/+landing/landing.component.js:52:3)
    Evaluating http://localhost:4200/app/+landing/landing.component.js
    Evaluating http://localhost:4200/app/+landing/index.js
    Evaluating http://localhost:4200/app/app.component.js
    Evaluating http://localhost:4200/app/index.js
    Evaluating http://localhost:4200/main.js
    Error loading http://localhost:4200/main.js ; Zone: <root> ; Task: Promise.then ; Value: Error: TypeError: Cannot read property 'forEach' of undefined(…)consoleError @ zone.js:461_loop_1 @ zone.js:490drainMicroTaskQueue @ zone.js:494ZoneTask.invoke @ zone.js:426

zone.js:463 Error: Uncaught (in promise): Error: TypeError: Cannot read property 'forEach' of undefined(…)
    consoleError        @   zone.js:463
    _loop_1             @   zone.js:490
    drainMicroTaskQueue @   zone.js:494
    ZoneTask.invoke     @   zone.js:426
```

--- a/src/polymer-element.spec.ts
+++ b/src/polymer-element.spec.ts
@@ -54,6 +54,18 @@ describe('PolymerElement', () => {
     }
   });
 
+  describe('Developer experience', () => {
+
+    it('should throw an error for non-registered elements', () => {
+      try {
+        PolymerElement('non-registered');
+      } catch (error) {
+        expect(error.message).toContain('element "non-registered" has not been registered');
+      }
+    });
+
+  });
+
   describe('Two-way data binding', () => {
 
     beforeAll(() => {

--- a/src/polymer-element.ts
+++ b/src/polymer-element.ts
@@ -72,6 +72,9 @@ export function PolymerElement(name: string): any[] {
   const arrayAndObjectProperties: Array<any> = [];
 
   const proto:any = Object.getPrototypeOf(document.createElement(name));
+  if (proto.is !== name) {
+    throw new Error(`The Polymer element "${name}" has not been registered. Please check that the element is imported correctly.`);
+  }
   const isFormElement:boolean = Polymer && Polymer.IronFormElementBehavior && proto.behaviors.indexOf(Polymer.IronFormElementBehavior) > -1;
   proto.behaviors.forEach((behavior:any) => configureProperties(behavior.properties));
   configureProperties(proto.properties);

